src/main/java/org/assertj/core/api/RecursiveIterableComparisonAssert.java

```diff
@@ -0,0 +1,285 @@
+   * For example, a List can be compared to a Set and even though they might have exact same elements, a Set will not preserve insertion order.
+   * So the assertion will fail. To avoid these kind of surprises, comparison ordered can be ignored with calling {@link #ignoringActualIterableOrder ignoringActualIterableOrder}.
+   * <p>
+   * Example:
+   * <pre><code class='java'>public class Person {
+   *     String name;
+   *     boolean hasPhd;
+   * }
+   *
+   * Person sheldon = new People("Sheldon Cooper", true);
+   * Person howard = new People("Howard Wolowitz", false);
+   * Person penny = new People("Penny", false);
+   *
+   * List<Person> peopleList = Arrays.asList(sheldon, howard, penny);
+   * Set<Person> peopleSet = new HashSet<>(peopleList);
+   *
+   * // assertion fails even though iterables contain the exact same elements because order is different.
+   * assertThat(peopleList).usingRecursiveComparison()
+   *                       .isEqualTo(peopleSet);
+   *
+   * // assertion succeeds because both iterables contain exact same items and comparison order is ignored.
+   * assertThat(peopleList).usingRecursiveComparison()
+   *                       .ignoringActualIterableOrder()
+   *                       .isEqualTo(peopleArray);</code></pre>
+   *
+   * @return this {@link RecursiveIterableComparisonAssert} to chain other methods.
+   */
+  public SELF ignoringActualIterableOrder() {
+    recursiveComparisonConfiguration.ignoreActualIterableOrder(true);
+    return myself;
+  }
+
+  /**
+   * By default, different type of iterables can be compared with each other. To change this behavior, use {@link #withStrictTypeCheckingOnActualIterable() withStrictTypeCheckingOnActualIterable}.
+   * This method restricts the types of iterables to be the same.
+   * <p>
+   * Example:
+   * <pre><code class='java'>public class Person {
+   *     String name;
+   *     boolean hasPhd;
+   * }
+   *
+   * Person sheldon = new People("Sheldon Cooper", true);
+   * Person howard = new People("Howard Wolowitz", false);
+   * Person penny = new People("Penny", false);
+   *
+   * List<Person> peopleList = Arrays.asList(sheldon, howard, penny);
+   * Set<Person> peopleSet = new LinkedHashSet<>(peopleList);
+   *
+   * // assertion succeeds because both iterables contain same items and both preserves insertion order.
+   * assertThat(peopleList).usingRecursiveComparison()
+   *                       .isEqualTo(peopleSet);
+   *
+   * // assertion fails because iterables types are restricted to be compatible, which in this case they are not.
+   * assertThat(peopleList).usingRecursiveComparison()
+   *                       .withStrictTypeCheckingOnActualIterable()
+   *                       .isEqualTo(peopleSet);</code></pre>
+   *
+   * @return this {@link RecursiveIterableComparisonAssert} to chain other methods.
+   */
+  public SELF withStrictTypeCheckingOnActualIterable() {
```

Comment:
For the same reason as the previous comment, I would rename it to `withStrictTypeCheckingOnIterable` or `withStrictIterableTypeChecking`.